app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.find_by(token: params[:id])
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+  end
+  
+  def create
+    user = User.find_by(token: params[:token])
+    if user
+      user.password = params[:password]
+      user.generate_token
+      user.save
```

Comment:
If you are not checking for a successful `save`, it is a good idea to use `save!`. Your user will be shown an error 500 page, but you will be notified of the exception so you can fix it. If you do it the current way, you will not know if there is a problem unless someone reports it.